crates/providers/src/new.rs

```diff
@@ -524,9 +739,23 @@ mod tests {
             gas: Some(U256::from(21000)),
             ..Default::default()
         };
+
+        let pending_tx = provider.send_transaction(tx.clone()).await.expect("failed to send tx");
+        let hash1 = *pending_tx.tx_hash();
```

Comment:
Deref + tx_hash() seems weird, would make this work with just derefing (bc it's convenient) or with `hash()`

Reply:
`impl Deref` is an anti-pattern and implicit copies make me sad, my opinion will not change on this :smiley:

Reply:
heh fair, not a strong one on my end, just don't love the `*x.foo()`, but yeah i hear the implicit copy point